### PR TITLE
add (helm "3.0") to Package-Requires

### LINKED ISSUE #499: 'slack.el' should declare dependency on 'helm'
**Describe the bug**

Because `helm-slack.el` has a dependency on the `helm` package, the elisp byte compiler throws errors for users who don't have helm.

**To Reproduce**
1. create a configuration that installs `slack` but not `helm`.
2. launch `emacs` and watch the messages (see below).

```
$ mkdir /tmp/debug
$ cd /tmp/debug
$ cat > debug-slack.el << EOF
(setq user-emacs-directory "/tmp/debug")
(require 'package)
(add-to-list 'package-archives '("melpa" . "https://melpa.org/packages/") t)
(setq package-enable-at-startup nil)
(package-initialize)
(package-refresh-contents)
(package-install 'slack)
EOF
$ emacs --batch -l ./debug-slack.el
< a bunch of output elided for brevity >
Compiling /tmp/debug/elpa/slack-20200115.308/helm-slack.el...

In toplevel form:
helm-slack.el:27:1:Error: Cannot open load file: No such file or directory, helm
```

**Backtrace**
n/a

**Expected behavior**
Package should install without compilation errors.

**Screenshots**
n/a

**Environment:**
 - OS: macOS Catalina, but this problem is OS independent
 - Emacs Version: 26.3
 - emacs-slack Version:

```
$ head elpa/slack-20200115.308/slack.el                                                                                                    ;;; slack.el --- slack client              -*- lexical-binding: t; -*-

;; Copyright (C) 2015  yuya.minami

;; URL: https://github.com/yuya373/emacs-slack
;; Author: yuya.minami <yuya.minami@yuyaminami-no-MacBook-Pro.local>
;; Keywords: tools
;; Version: 0.0.2
;; Package-Requires: ((websocket "1.8") (request "0.2.0") (circe "2.3") (alert "1.2") (emojify "0.4") (emacs "24.4"))
;; This program is free software; you can redistribute it and/or modify
```

**Additional context**

### PR DESCRIPTION
**Pull request by gonewest818**
Fix #499

It's not clear what version of `helm` is actually required, but since `helm-slack.el` was introduced in September 2018 and `helm` version 3.0 was released in August 2018, this seems like a reasonable choice.

Note that 99% of all the downloads of helm are from MELPA which continuously builds the latest commit from github (not MELPA stable which contains the version tagged releases). So it's likely many users are using versions of helm newer than 3.0 depending when they last updated their packages.

### CODE REVIEW

**Comment by yuya373** [comment]
I want to make helm as optional package.
Should I change `(require helm)` to `(declare-function some-function "helm")`?

**Comment by gonewest818** [comment]
Here's a discussion on reddit that might be helpful: https://www.reddit.com/r/emacs/comments/4dlk68/what_is_best_way_to_create_optional_library/

I think the correct answer is (ironically, but so typical for the internet these days) the one that got rated 0 points at the bottom of the page: "I think conditional dependencies should be considered an anti-pattern. If you have core functionality that doesn't depend on helm, then publish that. If you have optional functionality that does depend on helm, publish it separately. Conditional dependencies just lead to confusion..."

So in other words if you want to make the helm dependency optional then it's least confusing if you just publish `helm-slack` as a distinct package.

Otherwise you're going to be doing stuff conditionally (I haven't actually tested this)
```elisp
(if (featurep 'helm)             ; only if helm is present
  (with-no-warnings              ; suppress compiler errors for what follows

    ;; ... insert the entire contents of the file in here

  )
)
```

**Comment by yuya373** [comment]
It was very helpful. Thank you for the advice.
 I created [helm-slack](https://github.com/yuya373/helm-slack) as a separate package and remove dependency on helm from this package in https://github.com/yuya373/emacs-slack/pull/502

**Comment by yuya373** [comment]
I've merged #502, so close this pull request. Thank you!
